code/ContactList.java

```diff
@@ -57,22 +57,94 @@ private void expandCapacity() {
         friends = newFriends;
     }
 
-    public void remove(String firstName, String lastName) {
+    public int get(String firstName, String lastName) {
         // Create a temp friend object for easy
         // use of the Friend class' equals()
-        Friend toDelete = new Friend(firstName, lastName, "");
+        Friend toFind = new Friend(firstName, lastName, "");
 
-        // Linear search for the friend we are trying to delete
+        // Linear search for the friend we are trying to find
         for (int i = 0; i < friendCount; ++i) {
-            if (toDelete.equals(friends[i])) {
-                // Have friend at the end of the array be referenced
-                // by the array index we removed from
-                friends[i] = friends[friendCount - 1];
-                friends[friendCount - 1] = null;
-                friendCount--;
+            if (toFind.equals(friends[i])) {
+                return i;
             }
         }
+        // -1 will signify that we didn't find
+        // what we were looking for
+        return -1;
+    }
+
+
+//    public void remove(String firstName, String lastName) {
+//        // Create a temp friend object for easy
+//        // use of the Friend class' equals()
+//        Friend toDelete = new Friend(firstName, lastName, "");
+//
+//        // Linear search for the friend we are trying to delete
+//        for (int i = 0; i < friendCount; ++i) {
+//            if (toDelete.equals(friends[i])) {
+//                // Have friend at the end of the array be referenced
+//                // by the array index we removed from
+//                friends[i] = friends[friendCount - 1];
+//                friends[friendCount - 1] = null;
+//                friendCount--;
+//            }
+//        }
```

Comment:
This will remove every friend with the same full name. It's different behaviour than your re-implementation, which only removes the first occurrence.

Reply:
Good catch. 

In fact, I may leave this and include this point as a talking point.